src/diagonal.jl

```diff
@@ -1111,6 +1111,23 @@ end
 /(u::AdjointAbsVec, D::Diagonal) = (D' \ u')'
 /(u::TransposeAbsVec, D::Diagonal) = transpose(transpose(D) \ transpose(u))
 
+# norm
+function generic_normMinusInf(D::Diagonal)
+    norm_diag = norm(D.diag, -Inf)
+    min(norm_diag, zero(norm_diag))
```

Comment:
```suggestion
    return size(D,1) > 1 ? min(norm_diag, zero(norm_diag)) : norm_diag
```
a 1x1 diagonal matrix doesn't have a zero off-diagonal entry